src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static nature() {
+    this.alive = this.alive.filter((dead) =>
+      dead.health !== 0);
+  }
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100, toHide = false) {
+    super(name, health);
+    this.hidden = toHide;
+
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    Animal.alive.push(this);
+  }
+  bite(mammal) {
+    if (mammal.hidden === false) {
+      mammal.health -= 50;
+
+      if (mammal.health === 0) {
+        Animal.nature();
```

Comment:
Directly calling `Animal.nature()` after an animal's health reaches 0 within the `bite` method is not efficient. It would be better to perform this clean-up operation at a different time, such as after all biting has occurred, to avoid multiple calls which can be performance-intensive.